hcl2template/types.build.hcp_packer_registry.go

```diff
@@ -58,7 +64,33 @@ func (p *Parser) decodeHCPRegistry(block *hcl.Block) (*HCPPackerRegistryBlock, h
 
 	par.Slug = b.Slug
 	par.Description = b.Description
-	par.Labels = b.Labels
+
+	if len(b.Labels) > 0 && len(b.BucketLabels) > 0 {
+		diags = append(diags, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  fmt.Sprintf("%s.labels and %[1]s.bucket_labels are mutually exclusive; please use the recommended argument %[1]s.bucket_labels", buildHCPPackerRegistryLabel),
+			Subject:  block.DefRange.Ptr(),
+		})
+		return nil, diags
+	}
+
+	if len(b.Labels) > 0 {
+		diags = append(diags, &hcl.Diagnostic{
+			Severity: hcl.DiagWarning,
+			Summary:  fmt.Sprintf("the argument %s.labels has been deprecated and will be removed in a future release; please use %[1]s.bucket_labels", buildHCPPackerRegistryLabel),
```

Comment:
Should we be more precise here and say "in the next release" ?

Reply:
Good point. I made a change and included "minor" as that is usually when we remove deprecated things.

Reply:
Gotcha !